rollup.config.js

```diff
@@ -0,0 +1,252 @@
+ */
+function rollupConfig(input, outputFile, outputName, outputType, exports = 'auto', globals = {}, babelOptions = {}, resolverOptions = {}, external) {
+    return {
+        input: input,
+        external: external,
+        output: getOutput(outputFile, outputType, outputName, exports, globals),
+        plugins: getPlugins(outputType, babelOptions, resolverOptions)
+    };
+}
+
+/**
+ * Returns the correct output configuration
+ *
+ * @method getOutput
+ *
+ * @param {string} file
+ * @param {string} type
+ * @param {string} name
+ * @param {string} exports
+ * @param {object} globals
+ *
+ * @returns {{file: *, sourcemap: boolean, exports, format: *}|{file: *, sourcemap: boolean, exports: string, globals: *, name: *, format: string}}
+ */
+function getOutput(file, type, name, exports, globals) {
+    let config = {
+        exports: exports,
+        name: name,
+        file: file,
+        sourcemap: true,
+        format: type
+    };
+
+    if (type === 'umd') {
+        config.globals = globals;
+    }
+
+    return config;
+}
+
+/**
+ * Returns the correct plugins array
+ *
+ * @method getPlugins
+ *
+ * @param {string} type
+ * @param {object} babelOptions
+ * @param {object} resolverOptions
+ *
+ * @returns {Array}
+ */
+function getPlugins(type, babelOptions, resolverOptions) {
+    const babelPlugin = babel(
+        getBabelConfig(
+            babelOptions.exclude,
+            babelOptions.targets,
+            babelOptions.forceAllTransforms,
+            babelOptions.transformRuntimeOptions,
+            babelOptions.minified
+        )
+    );
+
+    switch (type) {
+        case 'umd':
+            return [
+                resolve(
+                    Object.assign(
+                        {
+                            browser: true,
+                            preferBuiltins: true
+                        },
+                        resolverOptions
+                    )
+                ),
+                commonjs(),
+                babelPlugin,
+                nodeGlobals(),
+                json(),
+                builtins(),
+                terser({sourcemap: true}),
+                bundleSize()
+            ];
+        case 'cjs':
+        case 'es':
+            return [
+                babelPlugin,
+                json(),
+                autoExternal(),
+                cleanup()
+            ];
+    }
+}
+
+
+/**
+ * Returns the correct babel config
+ *
+ * @method getBabelConfig
+ *
+ * @param {Array|string} exclude
+ * @param {object} targets
+ * @param {boolean} forceAllTransforms
+ * @param {object} transformRuntimeOptions
+ * @param {boolean} minified
+ *
+ * @returns {{minified: boolean, presets: *[][], babelrc: boolean, plugins: *[][], runtimeHelpers: boolean, exclude: *}}
+ */
+function getBabelConfig(exclude, targets, forceAllTransforms = false, transformRuntimeOptions = {}, minified = false) {
```

Comment:
I forgot to ask this. But why would you use this instead of babel's own config files? Does it have any advantage? Honest question, because I've never seen this.

Reply:
I could move the skeleton of this configuration out to a separate JSON file but I'm not sure if this will actually improve the readability.

Reply:
I'm ok with having the babel config in the rollup config. I just was curious about this.